lib/game.rb

```diff
@@ -110,4 +110,53 @@ def self.average_goals_per_season
       goals_per_season[season] = (total_goals / count_of_games_by_season[season].to_f).round(2)
     end
   end
+
+  def self.away_goals_per_team
+    away_goals = Hash.new(0)
+    all_games.each do |game|
+      away_goals[game.away_team_id] += game.away_goals
+    end
+    away_goals
+  end
+
+  def self.away_games_per_team
+    away_games = Hash.new(0)
+    all_games.each do |game|
+      away_games[game.away_team_id] += 1
+    end
+    away_games
+  end
+  
+  def self.away_avg_goals_per_team
+    avg_away_gpg = Hash.new(0)
+    self.away_goals_per_team.each do |team, goals|
+      avg_away_gpg[team] = (goals.to_f / Game.away_games_per_team[team]).round(2)
+    end
+    avg_away_gpg
+  end
+  
+  def self.highest_scoring_visitor
+    team_id_highest = self.away_avg_goals_per_team.max_by { |key, value| value }[0]
+    team_name_highest = nil
+    teams = TeamFactory.create_teams("./data/teams.csv")
+    Team.all_teams.each do |team_object|
+      if team_object.team_id == team_id_highest
+        team_name_highest = team_object.team_name
+      end
+    end
+    team_name_highest
+  end
+
+  def self.lowest_scoring_visitor
+    team_id_lowest = self.away_avg_goals_per_team.min_by { |key, value| value }[0]
+    team_name_lowest = nil
+    teams = TeamFactory.create_teams("./data/teams.csv")
+    Team.all_teams.each do |team_object|
+      if team_object.team_id == team_id_lowest
```

Comment:
Awesome, using the team id to eventually get the team name, which is the return value